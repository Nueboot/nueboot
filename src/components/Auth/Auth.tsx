import React from 'react';
import AuthUserContext from '../Session/AuthUserContext';

export const Login = () => <div>Login</div>;
export const Logout = () => <div>Logout</div>;

const Auth = () => (
  <AuthUserContext.Consumer>
    {authUser => authUser
      ? <Logout />
      : <Login />
    }
  </AuthUserContext.Consumer>
);

export default Auth;
