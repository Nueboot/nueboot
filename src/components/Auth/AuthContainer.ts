import { connect } from 'react-redux';

import { isLoggedIn } from 'state/Session/selectors';
import { loginUser, logoutUser } from '../../state/Session/actions';
import { LoginInfo } from '../../state/Session/types';
import Auth from './Auth';

export interface DispatchProps {
  loginUser(user: LoginInfo): void;
  signUpUser(user: LoginInfo): void;
  signout(): void;
}

export interface StateProps {
  loggedIn: boolean;
}

const mapStateToProps = (state, ownProps) => {
  return {
    loggedIn: isLoggedIn(state),
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    loginUser: (user: LoginInfo) => dispatch(loginUser('login', user)),
    signUpUser: (user: LoginInfo) => dispatch(loginUser('signup', user)),
    signout: () => dispatch(logoutUser()),
  };
};

const AuthContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Auth);

export default AuthContainer;
