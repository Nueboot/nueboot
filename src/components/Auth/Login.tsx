import React from 'react';
import styled from 'styled-components';

import { Dropdown, DropdownContent } from '../Styled/Dropdown';
import Text from '../Styled/Text';

const HoverText = styled(Text)`
  :hover {
    font-weight: 700;
  }
`;

const Login: React.SFC = () => (
  <Dropdown>
    <Text>Login ‚åµ</Text>
    <DropdownContent>
      <HoverText>Google</HoverText>
      <HoverText>Facebook</HoverText>
    </DropdownContent>
  </Dropdown>
);

export default Login;
