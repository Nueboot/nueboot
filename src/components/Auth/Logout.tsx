import React from 'react';
import styled from 'styled-components';

import { Dropdown, DropdownContent } from '../Styled/Dropdown';
import Text from '../Styled/Text';

interface LogoutProps {
  readonly displayName: string | null;
}

const HoverText = styled(Text)`
  :hover {
    font-weight: 700;
  }
`;

const Logout: React.SFC<LogoutProps> = ({ displayName }) => (
  <Dropdown>
    <Text>{displayName} ‚åµ</Text>
    <DropdownContent>
      <HoverText>Sign out</HoverText>
    </DropdownContent>
  </Dropdown>
);

export default Logout;
