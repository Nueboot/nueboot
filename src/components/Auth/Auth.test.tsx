const mockContext = jest.fn();
jest.mock('../Session/AuthUserContext', () => ({
  Consumer: ({ children }) => children(mockContext()),
}));

import { RouteComponentProps } from '@reach/router';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Auth, { Login, Logout } from './Auth';

describe('Auth', () => {
  let component: ShallowWrapper<RouteComponentProps>;

  describe('when no user is logged in', () => {
    beforeEach(() => {
      mockContext.mockReturnValue(null);
      component = shallow(<Auth />);
    });

    it('renders a login button', () => {
      expect(component.dive()).toMatchSnapshot();
    });

    describe('login component', () => {
      let loginComponent: ShallowWrapper;

      beforeEach(() => {
        loginComponent = shallow(<Login />);
      });

      it('renders correctly', () => {
        expect(loginComponent).toMatchSnapshot();
      });
    });
  });

  describe('when a user is logged in', () => {
    beforeEach(() => {
      mockContext.mockReturnValue({ authUser: { displayName: 'Tester' } });
      component = shallow(<Auth />);
    });

    it('renders a logout button', () => {
      expect(component.dive()).toMatchSnapshot();
    });

    describe('logout component', () => {
      let logoutComponent: ShallowWrapper;

      beforeEach(() => {
        logoutComponent = shallow(<Logout />);
      });

      it('renders correctly', () => {
        expect(logoutComponent).toMatchSnapshot();
      });
    });
  });

});
