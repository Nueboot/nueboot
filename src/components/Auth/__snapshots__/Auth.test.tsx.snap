// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth when a user is logged in logout component renders correctly 1`] = `
<div>
  Logout
</div>
`;

exports[`Auth when a user is logged in renders a logout button 1`] = `<Logout />`;

exports[`Auth when no user is logged in login component renders correctly 1`] = `
<div>
  Login
</div>
`;

exports[`Auth when no user is logged in renders a login button 1`] = `<Login />`;
